### PR TITLE
IA1-3996 howler quick changes for getting control over XHR and better…

### PR DESCRIPTION
**Pull request by jarlep**
In general Howler seems more robust in audio playback, rewind, seek etc. But it was lacking quite a lot on handling XHR requests, so the existing audioManager implementation was much more refined there.
To do minimum changes, the xhr object was made available to the outside, so progress and more refined error handling can be done. In addition stalled xhr requests can be aborted. This was all missing functionality in Howler.
I think making a contribution back to Howler with progress, abort functionality can be looked into.

### CODE REVIEW

**Comment by jarlep** [comment]
https://github.com/inspera/howler.js/pull/1/files#diff-1e099b745600c69cbaacbe86736975fc (src/howler.core.js) is the only real change. The other files are just dist changes (had to build and push the dist files)

**Review by hilhorstt — APPROVED** [review]
lgtm. Hope it does what it promises!

**Review by HansFredrik — COMMENTED** [review]
Will this enable us to track detailed load progress and report it to Monitor?

**Comment by jarlep** [comment]
> Will this enable us to track detailed load progress and report it to Monitor?

Yes, that part was reused from the "old" audio manager, as I have gotten access to the XHR object. I need that too for aborting hanging requests.

**Review by HansFredrik — APPROVED** [review]

File: src/howler.core.js [hunk 1]
```diff
@@ -2353,6 +2355,7 @@
     } else {
       // Load the buffer from the URL.
       var xhr = new XMLHttpRequest();
+      Howler.xhr = xhr;
```
Comment (asgeir-inspera) [inline src/howler.core.js:2358]: What do we achieve by this?

Comment (jarlep) [reply src/howler.core.js:2358]: What I write in the description. Improved error handling, progress and ability to abort request. I think we should consider implementing some of this in howler and contributing back, but this makes it possible to reuse some of the logic from the old audio manager (as it seemed the retrying and handling of loading worked pretty well there).

Comment (asgeir-inspera) [reply src/howler.core.js:2358]: I see the point after looking at the player PR. :-)

**Review by asgeir-inspera — APPROVED** [review]

**Pull request merged**